xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -117,6 +117,7 @@ SelectionStream& CSelectionStreams::Get(StreamType type, int index)
 
 std::vector<SelectionStream> CSelectionStreams::Get(StreamType type)
 {
+  CSingleLock lock(m_section);
```

Comment:
I understand, the only thing I don't understand is what you want me to do. I made three suggestions, and you didn't agree to any of these. I'd like to post something that meets your agreement, but it appears impossible.